cmd/libs/go2idl/openapi-gen/generators/openapi.go

```diff
@@ -0,0 +1,544 @@
+			}
+		}
+	}
+	switch len(targets) {
+	case 0:
+		// If no target package found, that means the generated file in target package is up to date
+		// and build excluded the target package.
+		return generator.Packages{}
+	case 1:
+		pkg := context.Universe[targets[0].Name.Package]
+		return generator.Packages{&generator.DefaultPackage{
+			PackageName: strings.Split(filepath.Base(pkg.Path), ".")[0],
+			PackagePath: pkg.Path,
+			HeaderText:  header,
+			GeneratorFunc: func(c *generator.Context) (generators []generator.Generator) {
+				return []generator.Generator{NewOpenAPIGen(arguments.OutputFileBaseName, pkg.Path, targets[0], context)}
+			},
+			FilterFunc: func(c *generator.Context, t *types.Type) bool {
+				// There is a conflict between this codegen and codecgen, we should avoid types generated for codecgen
+				if strings.HasPrefix(t.Name.Name, "codecSelfer") {
+					return false
+				}
+				pkg := context.Universe.Package(t.Name.Package)
+				if hasOpenAPITagValue(pkg.Comments, tagValueTrue) {
+					return !hasOpenAPITagValue(t.CommentLines, tagValueFalse)
+				}
+				if hasOpenAPITagValue(t.CommentLines, tagValueTrue) {
+					return true
+				}
+				return false
+			},
+		},
+		}
+	default:
+		glog.Fatalf("Duplicate target type found: %v", targets)
+	}
+	return generator.Packages{}
+}
+
+const (
+	specPackagePath       = "github.com/go-openapi/spec"
+	openAPIGenPackagePath = "k8s.io/kubernetes/cmd/libs/go2idl/openapi-gen/generators"
+)
+
+// openApiGen produces a file with auto-generated OpenAPI functions.
+type openAPIGen struct {
+	generator.DefaultGen
+	targetPackage string
+	imports       namer.ImportTracker
+	targetType    *types.Type
+	context       *generator.Context
+}
+
+func NewOpenAPIGen(sanitizedName, targetPackage string, targetType *types.Type, context *generator.Context) generator.Generator {
+	return &openAPIGen{
+		DefaultGen: generator.DefaultGen{
+			OptionalName: sanitizedName,
+		},
+		targetPackage: targetPackage,
+		imports:       generator.NewImportTracker(),
+		targetType:    targetType,
+		context:       context,
+	}
+}
+
+func (g *openAPIGen) Namers(c *generator.Context) namer.NameSystems {
+	// Have the raw namer for this file track what it imports.
+	return namer.NameSystems{
+		"raw": namer.NewRawNamer(g.targetPackage, g.imports),
+	}
+}
+
+func (g *openAPIGen) Filter(c *generator.Context, t *types.Type) bool {
+	// There is a conflict between this codegen and codecgen, we should avoid types generated for codecgen
+	if strings.HasPrefix(t.Name.Name, "codecSelfer") {
+		return false
+	}
+	return true
+}
+
+func (g *openAPIGen) isOtherPackage(pkg string) bool {
+	if pkg == g.targetPackage {
+		return false
+	}
+	if strings.HasSuffix(pkg, "\""+g.targetPackage+"\"") {
+		return false
+	}
+	return true
+}
+
+func (g *openAPIGen) Imports(c *generator.Context) []string {
+	importLines := []string{}
+	for _, singleImport := range g.imports.ImportLines() {
+		if g.isOtherPackage(singleImport) {
```

Comment:
I've added the change here. Is there any standard way to pull commits that are not merged yet into my PR? mine could (theoretically :) ) merge before that.